contracts/upgradeables/games/GReceipts.sol

```diff
@@ -0,0 +1,223 @@
+ *                   ,xNMMMMMMMMMMMMMMMMMMMMMMMWKl.        .lKWMMMMMMMMMMMMMMMMMMMMNk,
+ *                   .lXMMMMMMMMMMMMMMMMMMMMMMWKl.        'oKWMMMMMMMMMMMMMMMMMMMMMXl.
+ *                   ,kNMMMMMMMMMMMMMMMMMMMMMMNk;       ,kNMMMMMMMMMMMMMMMMMMMMMMNk,
+ *                   .;OWMMMMMMMMMMMMMMMMMMMMMMWOc.   .:OWMMMMMMMMMMMMMMMMMMMMMMW0;.
+ *                     .:0WMMMMMMMMMMMMMMMMMMMMMMWXd:,,dXWMMMMMMMMMMMMMMMMMMMMMMW0c.
+ *                      .:ONMMMMMMMMMMMMMMMMMMMMMMMNKK0NMMMMMMMMMMMMMMMMMMMMMMMNO:.
+ *                       .,dXWMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMWXx,.
+ *                         .:ONMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMNOc.
+ *                           .lONMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMNOl.
+ *                             .ckXWMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMWXkc.
+ *                               .;oOXNWMMMMMMMMMMMMMMMMMMMMMMMMMMMWNXOo;.
+ *                                 ..'cdOKNWMMMMMMMMMMMMMMMMMMMWNKOdc,..
+ *                                      ..;codkO00KXXXXKKK0Okdoc;..
+ *
+ */
+
+contract GReceipts is
+    Initializable,
+    AccessControlUpgradeable,
+    PausableUpgradeable,
+    ERC1155Upgradeable,
+    Achievo1155SoulboundUpgradeable,
+    ReentrancyGuardUpgradeable
+{
+    using SafeERC20 for IERC20;
+
+    event GUnitsSet(address indexed newGUnits);
+    event PaymentTokenSet(address indexed newPaymentToken);
+    error AddressIsZero();
+    error SoulboundError();
+
+    bytes32 public constant DEV_CONFIG_ROLE = keccak256("DEV_CONFIG_ROLE");
+    bytes32 public constant THIRD_PARTY_MINTER_ROLE =
+        keccak256("THIRD_PARTY_MINTER_ROLE");
+
+    address public gUnits;
+    address public paymentToken;
+
+    uint256 public constant DEFAULT_RECEIPT_ID = 1;
+
+    // @dev Initializes the contract
+    // @param _token The address of the token to use for the g-units
+    // @param _isPaused Whether the contract is paused
+    // @param _devWallet The address of the developer wallet
+    function initialize(
+        address _gUnits,
+        address _paymentToken,
+        bool _isPaused,
+        address _devWallet
+    ) public initializer {
+        __ReentrancyGuard_init();
+        __Pausable_init();
+        __AccessControl_init();
+        __ERC1155_init("");
+
+        if (
+            _devWallet == address(0) ||
+            _gUnits == address(0) ||
+            _paymentToken == address(0)
+        ) {
+            revert AddressIsZero();
+        }
+
+        _grantRole(DEFAULT_ADMIN_ROLE, _devWallet);
+        _grantRole(DEV_CONFIG_ROLE, _devWallet);
+        _setRoleAdmin(THIRD_PARTY_MINTER_ROLE, DEV_CONFIG_ROLE);
+
+        gUnits = _gUnits;
+        paymentToken = _paymentToken;
+
+        if (_isPaused) _pause();
+    }
+
+    function mint(
+        address _to,
+        uint256 _id,
+        uint256 _amount,
+        bytes memory _data
+    ) external onlyRole(DEV_CONFIG_ROLE) {
+        _mint(_to, _id, _amount, _data);

```

Comment:
you're missing the`_soulbound` usages

check you have an example from the level contract:

https://github.com/G7DAO/summon-contracts/blob/46593291fa621c472c4385452abbb74dd799af00/contracts/upgradeables/games/LevelsBoundV1.sol#L103C9-L103C19

Also the items